Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionContext.h

```diff
@@ -365,6 +369,8 @@ WK_CLASS_AVAILABLE(macos(13.3), ios(16.4))
  */
 - (BOOL)hasInjectedContentForURL:(NSURL *)url;
 
+@property (nonatomic, readonly) BOOL hasDynamicOrSessionDeclarativeNetRequestRules;
```

Comment:
This needs a comment block.

Reply:
This should be a more generic name, I would copy the one was have for `_WKWebExtension` with a `@ discussion `.

Something like:

```objc
/*!
 @abstract A boolean value indicating whether the extension includes rules used for content modification or blocking.
 @discussion This includes both static rules available in the extension's manifest and dynamic rules applied during a browsing session.

 */
@property (nonatomic, readonly) BOOL hasContentModificationRules;
```

We want the API to be as generic as possible so it can adapt as the underlying implementation might change as new web extension mechanisms are added.